microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/TracingDeploymentProcessor.java

```diff
@@ -46,32 +47,27 @@
 public class TracingDeploymentProcessor implements DeploymentUnitProcessor {
     @Override
     public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws DeploymentUnitProcessingException {
-        TracingExtensionLogger.ROOT_LOGGER.processingDeployment();
-
+        ROOT_LOGGER.processingDeployment();
         final DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
-        final CapabilityServiceSupport support = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
-
-        final WeldCapability weldCapability;
-        try {
-            weldCapability = support.getCapabilityRuntimeAPI(WELD_CAPABILITY_NAME, WeldCapability.class);
-        } catch (CapabilityServiceSupport.NoSuchCapabilityException e) {
-            //We should not be here since the subsystem depends on weld capability. Just in case ...
-            throw new DeploymentUnitProcessingException(TracingExtensionLogger.ROOT_LOGGER.deploymentRequiresCapability(
-                    deploymentPhaseContext.getDeploymentUnit().getName(),
-                    WELD_CAPABILITY_NAME
-            ));
-        }
-
         if (DeploymentTypeMarker.isType(DeploymentType.EAR, deploymentUnit)) {
             return;
         }
-
-        if (!weldCapability.isPartOfWeldDeployment(deploymentUnit)) {
-            // SmallRye JAX-RS requires CDI. Without CDI, there's no integration needed
-            TracingExtensionLogger.ROOT_LOGGER.noCdiDeployment();
-            return;
+        final CapabilityServiceSupport support = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
+        if (support.hasCapability(WELD_CAPABILITY_NAME)) {
```

Comment:
This check is confusing. Per org.wildfly.extension.microprofile.opentracing.SubsystemDefinition.OPENTRACING_CAPABILITY if this subsystem is present, WELD_CAPABILITY_NAME must be present too. So this if test cannot return false.  The try/catch at L57 is there because NoSuchCapabilityException is a checked exception and thus must be dealt with. Wrapping it with an if test implies that WELD_CAPABILITY_NAME might actually be an optional requirement, but it's not.